azurerm/internal/services/compute/windows_virtual_machine_resource.go

```diff
@@ -1245,6 +1385,29 @@ func resourceWindowsVirtualMachineDelete(d *schema.ResourceData, meta interface{
 		log.Printf("[DEBUG] Skipping Deleting OS Disk from Windows Virtual Machine %q (Resource Group %q)..", id.Name, id.ResourceGroup)
 	}
 
+	if features.VMDataDiskBeta() {
+		deleteDataDiskOnDeletion := meta.(*clients.Client).Features.VirtualMachine.DeleteDataDiskOnDeletion
+
+		// delete any disks created with the VM if feature toggled to do so. "existing" disks are not affected.
+		if deleteDataDiskOnDeletion {
+			if props := existing.VirtualMachineProperties; props != nil && props.StorageProfile != nil && props.StorageProfile.DataDisks != nil {
+				dataDisks := *props.StorageProfile.DataDisks
+				for _, v := range dataDisks {
+					if v.CreateOption == compute.DiskCreateOptionTypesEmpty && v.Name != nil {
+						deleteFuture, err := disksClient.Delete(ctx, id.ResourceGroup, *v.Name)
```

Comment:
this is slightly presumptuous - disks _should_ be in the same RG but don't have to be - perhaps we should parse the Disk ID and use those details?

Reply:
If the disk is created with the VM, it will always be in the same RG since there's no option to create it elsewhere? We don't touch `existing` disks that have been attached as part of the `deleteDataDiskOnDeletion` process.